### ISSUE TITLE
Error in `o2h` when hashref key has a leading `-`: `unknown option to o2h: '-1' at test.pl line 6.` (v 0.22)

### ISSUE THREAD

**Issue by oodler577** [haukex/Util-H2O#20]
Hi @haukex! I thought this was first an issue with my `Util::H2O:::o2d`, but see now it's `Util::H2O::o2h` - I have a hacky workaround for `o2d` in some coding where I am using it, but I have not been able to figure out the right way to deal with this, otherwise I'd provide a suggested PR. Thanks!

**Example 1**

```
use strict;
use warnings;
use Util::H2O qw/h2o o2h/;
my $foo = { foo => -1, code => sub { 1 } };
h2o $foo;
my $foo2 = o2h $foo;
```
Results in:

```
unknown option to o2h: '-1' at test.pl line 6.
```
Same result with this code, just using an accessor (`foo` to set value to `-1`):

**Example 2**

```
use strict;
use warnings;
use Util::H2O qw/h2o o2h/;
my $foo = { foo => 1, code => sub { 1 } };
h2o $foo;
my $foo2 = o2h $foo;

$foo->foo(-1);
$foo2 = o2h $foo;
```
Results in:

```
unknown option to o2h: '-1' at test.pl line 6.
```

**Comment by oodler577** [comment]
```
$ cpanm Util::H2O
Util::H2O is up to date. (0.22)
```

**Comment by haukex** [comment]
Thanks for finding this! I just released v0.24 that should fix this bug.
